infrastructure/zk/src/init.ts

```diff
@@ -240,6 +237,10 @@ async function checkEnv() {
     }
 }
 
+function additionalArgumentsForRedeployingL1(validium: boolean) {
+    return validium ? ['--validium-mode'] : [];
+}
```

Comment:
It is a bit unexpected that `init.ts` has to know arguments for a particular contracts script.
I think it'll be better to hold the arguments close to the script that uses them.
For example, you can change the signature of `redeployL1` to accept not only args, but also `validium: bool`. If it'll be `true`, you'll push an extra argument to the list of provided args.

Reply:
Also, to make it more typesafe you may introduce an enum, e.g. `DeploymentMode { Rollup, Validium }` to use instead of a boolean flag.

Reply:
Done in [e1e01dd](https://github.com/matter-labs/zksync-era/pull/1242/commits/e1e01dd55a9a86ebabc06e0b1737d0c2cccab83f)